### PR TITLE
Inital addition of aws-nuke workflows.

### PR DESCRIPTION
**Pull request by justin-o12**
This initial commit will run everyday at 1 AM Mountain and targets the following for deletion in the CI account.
- S3Object
- S3Bucket
- EC2Instance
- EKSNodegroups
- EKSCluster

### CODE REVIEW

File: .github/workflows/aws-nuke/aws-nuke-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.AWS_COMMERCIAL_ROLE_TO_ASSUME }}
+          role-session-name: ${{ github.job || github.event.client_payload.pull_request.head.sha || github.sha }}
+          aws-region: us-west-2
+          role-duration-seconds: 21600
+
+      - name: Download aws-nuke
+        run: wget -c https://github.com/rebuy-de/aws-nuke/releases/download/v2.23.0/aws-nuke-v2.23.0-linux-amd64.tar.gz -O - | tar xz
```
Comment (lucasrod16) [inline .github/workflows/aws-nuke/aws-nuke-ci.yaml:20]: We may want to configure renovate to watch `aws-nuke` for automated updates

```suggestion
        # renovate: datasource=github-tags depName=rebuy-de/aws-nuke versioning=semver
        run: wget -c https://github.com/rebuy-de/aws-nuke/releases/download/v2.23.0/aws-nuke-v2.23.0-linux-amd64.tar.gz -O - | tar xz
```

File: .github/workflows/aws-nuke/aws-nuke-ci.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+      - name: Run aws-nuke
+        run: ./aws-nuke-v2.23.0-linux-amd64 --config ./aws-nuke-config.yaml
```
Comment (lucasrod16) [inline .github/workflows/aws-nuke/aws-nuke-ci.yaml:23]: According the [`aws-nuke` docs](https://github.com/rebuy-de/aws-nuke#caution), the `--no-dry-run` flag is required for it to actually delete resources. Maybe we want to have a "dry-run" stage and a "live" stage?

```suggestion
        run: ./aws-nuke-v2.23.0-linux-amd64 --config ./aws-nuke-config.yaml --no-dry-run
```

Comment (justin-o12) [reply .github/workflows/aws-nuke/aws-nuke-ci.yaml:24]: How would the dry-run and live stages work here?  Also, I suggest merging this without `--no-dry-run` for now to see it run on its own first and then enabling it with a follow-on PR.

**Comment by justin-o12** [comment]
[filter-nuke.txt](https://github.com/defenseunicorns/uds-package-dubbd/files/11972130/filter-nuke.txt)

Long list of resources that would not be deleted.  These are mostly state files the two state file buckets.

File: .github/workflows/aws-nuke/aws-nuke-ci-config.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+regions:
+  - global
+  - us-east-2
+  - us-east-1
+  - us-west-1
+  - us-west-2
+
+account-blocklist:
+  - "999999999999" # production
```
Comment (justin-o12) [inline .github/workflows/aws-nuke/aws-nuke-ci-config.yaml:9]: The tool complains if we don't get it an account-blocklist.

File: .github/workflows/aws-nuke/aws-nuke-ci.yaml (hunk 2 above)
Comment (lucasrod16) [reply .github/workflows/aws-nuke/aws-nuke-ci.yaml:24]: I was thinking the dry-run stage would just be a step in the job prior to running the next step that uses the `--no-dry-run` flag to actually delete the resources. Similar in concept to running `terraform plan` prior to `terraform apply`. Not sure whether that scenario would be helpful for us or not in this use case. Moving forward without `--no-dry-run` sounds good to me.

**Review by lucasrod16 — APPROVED** [review]
LGTM

**Pull request merged**